### ISSUE TITLE
aws_kinesis_firehose_delivery_stream for cross-account ElasticSearch

### ISSUE THREAD

**Issue by lapkritinis** [terraform-providers/terraform-provider-aws#14915]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* aws_kinesis_firehose_delivery_stream

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Actual Behavior

1 error occurred:
        * aws_kinesis_firehose_delivery_stream.waf-mgmt-es: 1 error occurred:
        * aws_kinesis_firehose_delivery_stream.waf-mgmt-es: error creating Kinesis Firehose Delivery Stream: InvalidArgumentException: Domain not found: my-es-domain-public

### Steps to Reproduce

The issue is present when I try to create a cross-account firehose delivery stream to ElasticSearch.
Reason for that I believe, that via AWS CLI when I create this delivery stream I specify an additional parameter:
ClusterEndpoint - which is not supported in terraform and I do not specify domain.

resource "aws_kinesis_firehose_delivery_stream" "waf-mgmt-es" {
  destination = "elasticsearch"
  name = "aws-waf-logs-mgmt-es"

  elasticsearch_configuration {
    domain_arn = "arn:aws:es:eu-central-1:<--accountB-->:domain/my-es-domain-public"
    index_name = "waf-${var.env_name}"
    role_arn = "arn:aws:iam::<--accountA-->:role/firehose_role"
  }

  s3_configuration {
    role_arn           = "arn:aws:iam::991799896228:role/firehose_temp_role"
    bucket_arn         = "${aws_s3_bucket.temp.arn}"
    buffer_size        = 10
    buffer_interval    = 400
    compression_format = "GZIP"
  }
}

1. `terraform apply`

### References

* #0000

**Comment by ewbankkit** [comment]
We need to add `cluster_endpoint` to `elasticsearch_configuration` configuration block.

**Comment by ewbankkit** [comment]
@lapkritinis Thanks for raising this issue.
It has already been noticed in #10623 and is resolved with #12484.
 I'm going to close this one as a duplicate so that we can concentrate discussion in the linked issue.
Please add any additional comments there.
